README.md

```diff
@@ -194,8 +281,11 @@ Reference the [Dagger-Hilt testing package](https://dagger.dev/hilt/testing) for
 
 # Limitations
 
-Pommel currently only supports being annotated on a `class`, as a result it assumes that the `class` has at least one constructor that is annotated with `@Inject` and
-that all parameters passed into the constructor are also on the Dagger graph.
+Pommel expects `class`es annotated with `@SoloModule` to have a constructor that is is annotated with `@Inject`. Multiple constructors annotated with `@Inject` is not supported as Dagger itself only supports a single
+constructor annotated with `@Inject`. All parameters passed into the constructor must also be on the Dagger graph, as this is another expectation of Dagger constructor injection
+
+Pommel expects all functions annotated with `@SoloModule` to be static due to the fact the generated module needs to call the annotated function.
+As a result Pommel currently only supports top level functions and functions in a top level `object` that are annotated with `@JvmStatic`. Although functions within a companion object are static they are currently not supported.
```

Comment:
```suggestion
As a result Pommel currently only supports top level functions and functions in a top level `object` that are annotated with `@JvmStatic`. `static` functions within a `companion object` are currently not supported.
```

Reply:
You had the word "currently" in there; is support for `static` functions within a `companion object` planned?

Reply:
@twyatt I think makes sense to support all Kotlin static functions. Though the more I think about it would be a bit weird to have a something like the following 

```kotlin
class  MyClass @Inject constructor(@Named("name") private val name: String) {
    companion object {
        @SoloModule
        fun name() = "pommel"
    }
}
```

not sure why someone would want to have their Dagger configuration in a companion object but whom am I to judge. I also plan to eliminate the need of using the `@JvmStatic` annotation

Reply:
I think a more common case for a companion object function probably looks like this:

```kotlin
class MyClass private constructor(private val name: String) {
    companion object {
        @SoloModule
        fun create(
            @Named("name") name: String
        ) = MyClass(name)
    }
}
```